src/util/Thread.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright 2019 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "util/Thread.h"
+#include "util/Logging.h"
+
+#ifdef _WIN32
+#else
+#include <unistd.h>
+#endif
+
+namespace stellar
+{
+
+#ifdef _WIN32
+
+void
+runCurrentThreadWithLowPriority()
+{
+    auto native = std::this_thread::get_id();
+    auto ret = SetThreadPriority(native, THREAD_PRIORITY_BELOW_NORMAL);
+
+    if (!ret)
+    {
+        CLOG(DEBUG, "Fs") << "Unable to set priority for thread: " << ret;
+    }
+}
+
+#else
+
+void
+runCurrentThreadWithLowPriority()
+{
+    constexpr auto const LOW_PRIORITY_NICE = 5;
+
+    auto newNice = nice(LOW_PRIORITY_NICE);
```

Comment:
Check with somebody else, but I think this will not work for BSD (MacOS) and just renice the entire process.

Maybe MacOS doesn't allow to do "low priority threads" - there is some conflicting information on [their site](https://developer.apple.com/library/archive/documentation/Darwin/Conceptual/KernelProgramming/scheduler/scheduler.html#//apple_ref/doc/uid/TP30000905-CH211-BABFGDHA) where it seems threads at at least "normal priority" (but later, there is mention of tasks that can have `TASK_BACKGROUND_APPLICATION` (but I don't know if a task maps to a process, thread or something else).

Maybe we don't do anything on MacOS (but then we may be breaking FreeBSD as well in general)

Reply:
I've moved nice to be only used on linux. Also added comment that maybe thread_policy_set with THREAD_PRECEDENCE_POLICY can be used to set low priority on MacOS (someone with MacOS must verify that) - it allows setting to IDLE_PRI, so probably some intermediate lower priority setting are also available.

And from the documentation I think that TASK_BACKGROUND_APPLICATION sets priority for whole process, not a thread.